[TC-7427] Add retry strategies (#41)
tradecloud/tradecloud-docs-api-v2: api/rules.md

The **account number should be set on forehand** in the **Tradecloud connection** with your supplier or buyer. You can set the account code when inviting a new connection or at any time in the connection overview in the portal.

## Sending messages

Rules for any Tradecloud API request:

### Send messages sequential

Send the next message when the previous one has been responded with HTTP Status Code 200 OK or 202 Accepted.

Do **not** send more than **10 requests per second**.

Tradecloud may respond with [HTTP Status Code 429](https://tools.ietf.org/html/rfc6585#section-4) `Too Many Requests`

### Queue and retry messages with an exponential backoff.

> The Tradecloud [environments](environments.md) are not 100% available. It is the API client's responsibility to queue the message and automatically retry with exponential backoff till Tradecloud is available again. An alternative is to warn the ERP user, who is trying to send the message to Tradecloud, so the user can manually retry later.

Reviewer: ```suggestion
The Tradecloud [environments](environments.md) do not have an availability SLO of 100%. If Tradecloud is temporarily unavailable, it is the API client's responsibility to queue the message and automatically retry with exponential backoff till Tradecloud is available again. An alternative is to warn the ERP user, who is trying to send the message to Tradecloud, so the user can manually retry later.
```

> When Tradecloud responds with either a: 
> 
> - any HTTP Status Code 5xx Server Error or
> - [HTTP Status Code 429](https://tools.ietf.org/html/rfc6585#section-4) `Too Many Requests` or 
> - the requests does time out (currently 5 secs. at Tradecloud API side)
>   
> Then the client must automatically retry using an exponential backoff strategy or use a manual retry strategy. 
> The [Exponential Backoff Calculator](http://backoffcalculator.com/?interval=5&rate=2&attempts=5) is handy to verify the retry plan.
> 

Reviewer: ```suggestion
If Tradecloud responds with either: 

- a HTTP Status Code 5xx Server Error or
- [HTTP Status Code 429](https://tools.ietf.org/html/rfc6585#section-4) `Too Many Requests` or 
- the requests does time out (currently 5 secs. at Tradecloud API side)
  
Then the client must automatically retry, using an exponential backoff strategy, or use a manual retry strategy. 
The [Exponential Backoff Calculator](http://backoffcalculator.com/?interval=5&rate=2&attempts=5) is handy to verify the retry plan.

```
